In [6]:
import load
import pandas as pd
import numpy as np

In [18]:
tables = load.all_tables()
print(tables.keys())

dict_keys(['av_patient', 'av_tumour', 'sact_cycle', 'sact_drug_detail', 'sact_outcome', 'sact_patient', 'sact_regimen', 'sact_tumour'])


In [27]:
merged_av = pd.merge(tables['av_patient'], tables['av_tumour'], on='PATIENTID', how='left')
print(merged_av.columns)

Index(['PATIENTID', 'SEX_x', 'LINKNUMBER_x', 'ETHNICITY', 'DEATHCAUSECODE_1A',
       'DEATHCAUSECODE_1B', 'DEATHCAUSECODE_1C', 'DEATHCAUSECODE_2',
       'DEATHCAUSECODE_UNDERLYING', 'DEATHLOCATIONCODE', 'NEWVITALSTATUS',
       'VITALSTATUSDATE', 'TUMOURID', 'DIAGNOSISDATEBEST', 'SITE_ICD10_O2',
       'SITE_ICD10_O2_3CHAR', 'MORPH_ICD10_O2', 'BEHAVIOUR_ICD10_O2', 'T_BEST',
       'N_BEST', 'M_BEST', 'STAGE_BEST', 'STAGE_BEST_SYSTEM', 'GRADE', 'AGE',
       'SEX_y', 'CREG_CODE', 'LINKNUMBER_y', 'SCREENINGSTATUSFULL_CODE',
       'ER_STATUS', 'ER_SCORE', 'PR_STATUS', 'PR_SCORE', 'HER2_STATUS',
       'CANCERCAREPLANINTENT', 'PERFORMANCESTATUS', 'CNS', 'ACE27',
       'GLEASON_PRIMARY', 'GLEASON_SECONDARY', 'GLEASON_TERTIARY',
       'GLEASON_COMBINED', 'DATE_FIRST_SURGERY', 'LATERALITY',
       'QUINTILE_2015'],
      dtype='object')


In [28]:
print(np.all(merged_av['SEX_x'] == merged_av['SEX_y']))
print(np.all(merged_av['LINKNUMBER_x'] == merged_av['LINKNUMBER_y']))

True
True


In [34]:
print(len(merged_av))
print(len(tables['av_patient']))
print(len(tables['av_tumour']))
print(len(merged_av.PATIENTID.unique()))
print(len(merged_av.TUMOURID.unique()))

1402817
1322100
1402817
1322100
1402817


Numbers check out

In [29]:
merged_sact = pd.merge(tables['sact_patient'], tables['sact_tumour'], on='MERGED_PATIENT_ID', how='left')
print(merged_sact.columns)

Index(['MERGED_PATIENT_ID', 'LINK_NUMBER', 'MERGED_TUMOUR_ID',
       'CONSULTANT_SPECIALITY_CODE', 'PRIMARY_DIAGNOSIS', 'MORPHOLOGY_CLEAN'],
      dtype='object')


In [40]:
print(len(merged_sact))
print(len(tables['sact_patient']))
print(len(tables['sact_tumour']))
print(len(merged_sact.MERGED_PATIENT_ID.unique()))
print(len(merged_sact.MERGED_TUMOUR_ID.unique()))
print(merged_sact.MERGED_TUMOUR_ID.count())

355312
245938
299727
245938
299723
299722


unique Tumour Ids : 299723 count: 299722 meaning one of those is NaN  
245938 unique Patient Ids (none will be NaN)  
355312 rows  
355312-299722=55590 patients with no tumour  
299727-299722=5 tumours with no patient

In [56]:
total_merged = pd.merge(merged_av, merged_sact, left_on='LINKNUMBER_x', right_on='LINK_NUMBER', how='inner')
print(total_merged.columns)


Index(['PATIENTID', 'SEX_x', 'LINKNUMBER_x', 'ETHNICITY', 'DEATHCAUSECODE_1A',
       'DEATHCAUSECODE_1B', 'DEATHCAUSECODE_1C', 'DEATHCAUSECODE_2',
       'DEATHCAUSECODE_UNDERLYING', 'DEATHLOCATIONCODE', 'NEWVITALSTATUS',
       'VITALSTATUSDATE', 'TUMOURID', 'DIAGNOSISDATEBEST', 'SITE_ICD10_O2',
       'SITE_ICD10_O2_3CHAR', 'MORPH_ICD10_O2', 'BEHAVIOUR_ICD10_O2', 'T_BEST',
       'N_BEST', 'M_BEST', 'STAGE_BEST', 'STAGE_BEST_SYSTEM', 'GRADE', 'AGE',
       'SEX_y', 'CREG_CODE', 'LINKNUMBER_y', 'SCREENINGSTATUSFULL_CODE',
       'ER_STATUS', 'ER_SCORE', 'PR_STATUS', 'PR_SCORE', 'HER2_STATUS',
       'CANCERCAREPLANINTENT', 'PERFORMANCESTATUS', 'CNS', 'ACE27',
       'GLEASON_PRIMARY', 'GLEASON_SECONDARY', 'GLEASON_TERTIARY',
       'GLEASON_COMBINED', 'DATE_FIRST_SURGERY', 'LATERALITY', 'QUINTILE_2015',
       'MERGED_PATIENT_ID', 'LINK_NUMBER', 'MERGED_TUMOUR_ID',
       'CONSULTANT_SPECIALITY_CODE', 'PRIMARY_DIAGNOSIS', 'MORPHOLOGY_CLEAN'],
      dtype='object')


In [57]:
print(len(total_merged))

393921


In [66]:
print(sum(total_merged['SITE_ICD10_O2'].astype(str) == total_merged['PRIMARY_DIAGNOSIS'].astype(str)))
print(sum(total_merged['SITE_ICD10_O2_3CHAR'].astype(str) == total_merged['PRIMARY_DIAGNOSIS'].astype(str)))


90920
31797


__**This is small (about 1/4) indicating they are not related in the simulacrum data set**__




   


In [81]:
diagnosis_df = total_merged[["PRIMARY_DIAGNOSIS", "MORPHOLOGY_CLEAN","SITE_ICD10_O2","SITE_ICD10_O2_3CHAR",
              "MORPH_ICD10_O2","BEHAVIOUR_ICD10_O2"]]
diagnosis_df

PRIMARY_DIAGNOSIS MORPHOLOGY_CLEAN SITE_ICD10_O2 SITE_ICD10_O2_3CHAR  \
0                    NaN              NaN          C444                 C44   
1                    NaN              NaN          C679                 C44   
2                    C61            81403          C449                 C44   
3                    C61              NaN          C443                 C44   
4                    C61              NaN          C252                 C83   
5                   C679            81403          C449                 C44   
6                   C500              NaN          C443                 C44   
7                   C155              NaN          C449                 C44   
8                   C155              NaN          C187                 C02   
9                   C159              NaN          C445                 C44   
10                  C159              NaN          C679                 D09   
11                  C829              NaN          C445                 C44   
12                  D473              NaN          C443                 C44   
13                  C833            96803          C443                 C44   
14                  C440              NaN          C443                 C44   
15                  C504            80703          C443                 C44   
16                  C504            80703          C443                 C43   
17                   NaN              NaN          C447                 C44   
18                   NaN              NaN          C187                 D01   
19                   C61            81403          C447                 C44   
20                  C109              NaN          C443                 C44   
21                  C109              NaN          C671                 C61   
22                  C159              NaN          C449                 C44   
23                  C341            81403          C449                 C44   
24                  C159              NaN          C182                 C20   
25                  C341            81403          C182                 C20   
26                   C61            81403          C449                 C44   
27                  D471            99623          C449                 C44   
28                  D471            99623           C61                 C16   
29                  C920              NaN          C446                 C44   
...                  ...              ...           ...                 ...   
393891              C439              NaN          C470                 C47   
393892               NaN              NaN          C474                 C47   
393893              C348              NaN          C944                 C94   
393894              C865            97183          C930                 C93   
393895              C865            97183          D002                 D09   
393896               C55            94213          C942                 C94   
393897              C419              NaN          C701                 C58   
393898               C80            88003          C940                 C94   
393899              C482            81403          C940                 C94   
393900               C80            88003          C341                 C44   
393901              C482            81403          C341                 C44   
393902              C414            84413          C473                 C47   
393903              C845              NaN          C473                 C47   
393904              O010              NaN           C33                 C33   
393905               NaN              NaN          C940                 C94   
393906              C830            97323          C930                 C93   
393907              C920            80703          C930                 C93   
393908              C601            96633          C930                 C93   
393909              C601            96633          C679  

In [68]:
merged_regimen = pd.merge(merged_sact, tables['sact_regimen'], on='MERGED_TUMOUR_ID', how='left')


In [70]:
print(len(merged_regimen))
print(len(tables['sact_regimen']))
print(len(merged_regimen.MERGED_REGIMEN_ID.unique()))
print(merged_regimen.MERGED_REGIMEN_ID.count())

527502
471919
471913
471912


527502-471912-55590=0 tumours with no regimen  
471919-471912=7 regimens with no tumour (might be connected to the 5 tumours with no patient)

In [72]:
merged_outcome = pd.merge(merged_regimen, tables['sact_outcome'], on='MERGED_REGIMEN_ID', how='left')


In [74]:
print(len(merged_outcome))
print(len(tables['sact_outcome']))
print(len(merged_outcome.MERGED_OUTCOME_ID.unique()))
print(merged_outcome.MERGED_OUTCOME_ID.count())

527502
351668
351662
351661


All regimens have at most 1 outcome (as it should be)  
7 outcomes with no regimen (probably connected to the 7 regimens with no tumour)
527502-351661=lots of regimens with no outcome